Branch-SDK/src/io/branch/referral/ServerRequestRegisterView.java

```diff
@@ -0,0 +1,147 @@
+ * * <p>
+ * The server request for registering a view event of a specific content specified by user given attributes
+ * </p>
+ */
+class ServerRequestRegisterView extends ServerRequest {
+
+    BranchUniversalObject.RegisterViewStatusListener callback_;
+
+    /**
+     * <p>Create an instance of {@link ServerRequestRegisterView} to notify Branch on a content view event.</p>
+     *
+     * @param branchUniversalObject An instance of {@link BranchUniversalObject} to mar as viewed
+     */
+    public ServerRequestRegisterView(Context context, BranchUniversalObject branchUniversalObject, SystemObserver sysObserver, BranchUniversalObject.RegisterViewStatusListener callback) {
+        super(context, Defines.RequestPath.RegisterView.getPath());
+        callback_ = callback;
+        JSONObject registerViewPost;
+        try {
+            registerViewPost = createContentViewJson(branchUniversalObject, sysObserver);
+            setPost(registerViewPost);
+        } catch (JSONException ex) {
+            ex.printStackTrace();
+            constructError_ = true;
+        }
+    }
+
+    @Override
+    public void onRequestSucceeded(ServerResponse resp, Branch branch) {
+        if (callback_ != null) {
+            callback_.onRegisterViewFinished(true, null);
+        }
+    }
+
+    @Override
+    public void handleFailure(int statusCode) {
+        if (callback_ != null) {
+            callback_.onRegisterViewFinished(false, new BranchError("Unable to register content view", statusCode));
+        }
+    }
+
+    @Override
+    public boolean handleErrors(Context context) {
+        if (!super.doesAppHasInternetPermission(context)) {
+            callback_.onRegisterViewFinished(false, new BranchError("Unable to register content view", BranchError.ERR_NO_INTERNET_PERMISSION));
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isGetRequest() {
+        return false;
+    }
+
+    @Override
+    public void clearCallbacks() {
+        callback_ = null;
+    }
+
+    /**
+     * Creates a Json with given parameters for register view.
+     *
+     * @param universalObject An instance of {@link BranchUniversalObject} to create the Json
+     * @return A {@link JSONObject} for post data for register view request
+     * @throws JSONException {@link JSONException} on any Json errors
+     */
+    private JSONObject createContentViewJson(BranchUniversalObject universalObject,
+                                             SystemObserver sysObserver) throws JSONException {
+
+        JSONObject contentObject = new JSONObject();
+
+
+        String os_Info = "Android " + Build.VERSION.SDK_INT;
+        String sessionID = prefHelper_.getSessionID();
+
+        contentObject.put(Defines.Jsonkey.OS.getKey(), os_Info);
+        contentObject.put(Defines.Jsonkey.SessionID.getKey(), sessionID);
+        contentObject.put(Defines.Jsonkey.DeviceFingerprintID.getKey(), prefHelper_.getDeviceFingerPrintID());
```

Comment:
the server handles the fact that device_fingerprint_id is bnc_no_value, right? in cases where a `registerView` is called very soon after a `v1/install` call